pkg/config/net.go

```diff
@@ -34,13 +39,36 @@ func GetInterfaceAndNonVIPAddr(vips []net.IP) (vipIface net.Interface, nonVipAdd
 				_, nn, _ := net.ParseCIDR(strings.Replace(addr.String(), "/128", "/64", 1))
 
 				if nn.Contains(vips[0]) {
-					return iface, n, err
+					return iface, n, nil
 				}
 			default:
 				fmt.Println("not supported addr")
 			}
 		}
 	}
 
-	return vipIface, nonVipAddr, fmt.Errorf("No interface nor address found for the given VIPs")
+	nodeAddrs, err := utils.AddressesDefault(utils.ValidNodeAddress)
+	if err != nil {
+		return vipIface, nonVipAddr, err
+	}
+	if len(nodeAddrs) == 0 {
+		return vipIface, nonVipAddr, fmt.Errorf("No interface nor address found")
+	}
+	for _, iface := range ifaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return vipIface, nonVipAddr, err
+		}
+		for _, addr := range addrs {
+			switch n := addr.(type) {
+			case *net.IPNet:
+				if n.IP.String() == nodeAddrs[0].String() {
```

Comment:
I'm curious what would happen if AddressesDefault returned multiple matches, but I guess this only returns one address anyway so it probably doesn't matter.

Reply:
AddressesDefault could return multiple matches if there are more than one interface configured on the subnet with the default routes. In this case we pick the first match, consistent with [how node-ip is implemented](https://github.com/openshift/baremetal-runtimecfg/blob/1cd2a95/cmd/runtimecfg/node-ip.go#L135). Do you think there's is a risk AddressesDefault returns results in a different order?

Reply:
I'm not sure. It's just that when I see something return an array and then the calling code only uses the first element it makes me wonder if there could be a bug hidden in there under some circumstances.

It's an existing problem with the function though. We're returning exactly one ip from calls that could return multiple in the existing code too, so I'm not worried about it in the context of this change. It was sort of a general observation.